controller/src/main/java/io/pravega/controller/server/security/auth/handler/impl/AclAuthorizerImpl.java

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.security.auth.handler.impl;
+
+import io.pravega.auth.AuthHandler;
+import lombok.NonNull;
+
+class AclAuthorizerImpl extends AclAuthorizer {
+
+    @Override
+    public AuthHandler.Permissions authorize(@NonNull AccessControlList accessControlList, @NonNull String resource) {
+        AuthHandler.Permissions result = AuthHandler.Permissions.NONE;
+
+        String resourceDomain;
+        if (resource.indexOf("::") > 0) {
```

Comment:
You're doing a double-lookup here. Get the index of `"::"` and check for negative values, then reuse it in your `substring` call.

Reply:
Yeah. I've now modified the code to ensure that is done once.